src/libraries/Microsoft.Extensions.Configuration.Binder/tests/SourceGenerationTests/Baselines/TestPrimitivesGen.generated.txt

```diff
@@ -0,0 +1,178 @@
+        {
+            if (obj is null)
+            {
+                throw new ArgumentNullException(nameof(obj));
+            }
+
+            if (configuration["MyString"] is string stringValue1)
+            {
+                obj.MyString = stringValue1;
+            }
+
+            if (configuration["MyInt128"] is string stringValue2)
+            {
+                try
+                {
+                    obj.MyInt128 = Int128.Parse(stringValue2, CultureInfo.InvariantCulture);
+                }
+                catch (Exception exception)
+                {
+                    throw new InvalidOperationException($"Failed to convert configuration value at '{configuration.GetSection("MyInt128").Path}' to type '{typeof(Int128)}'.", exception);
+                }
+            }
+
+            if (configuration["MyInt"] is string stringValue3)
+            {
+                try
+                {
+                    obj.MyInt = int.Parse(stringValue3, NumberStyles.Integer, CultureInfo.InvariantCulture);
+                }
+                catch (Exception exception)
+                {
+                    throw new InvalidOperationException($"Failed to convert configuration value at '{configuration.GetSection("MyInt").Path}' to type '{typeof(int)}'.", exception);
+                }
+            }
+
+            if (configuration["MyUInt128"] is string stringValue4)
+            {
+                try
+                {
+                    obj.MyUInt128 = UInt128.Parse(stringValue4, CultureInfo.InvariantCulture);
+                }
+                catch (Exception exception)
+                {
+                    throw new InvalidOperationException($"Failed to convert configuration value at '{configuration.GetSection("MyUInt128").Path}' to type '{typeof(UInt128)}'.", exception);
+                }
+            }
+
+            if (configuration["MyLong"] is string stringValue5)
+            {
+                try
+                {
+                    obj.MyLong = long.Parse(stringValue5, NumberStyles.Integer, CultureInfo.InvariantCulture);
+                }
+                catch (Exception exception)
+                {
+                    throw new InvalidOperationException($"Failed to convert configuration value at '{configuration.GetSection("MyLong").Path}' to type '{typeof(long)}'.", exception);
+                }
+            }
+
+            if (configuration["MyUri"] is string stringValue6)
+            {
+                Uri.TryCreate(stringValue6, UriKind.RelativeOrAbsolute, out Uri? temp7);
+                if (temp7 is not null)
+                {
+                    obj.MyUri = temp7;
+                }
+            }
+
+            if (configuration["MyCultureInfo"] is string stringValue8)
+            {
+                try
+                {
+                    obj.MyCultureInfo = CultureInfo.GetCultureInfoByIetfLanguageTag(stringValue8);
```

Comment:
There's a bunch more logic in the CultureInfoConverter than this:

[https://github.com/dotnet/runtime/blob/5711ca33f8dd1758e18c09ebdedf7e95eb424e0f/src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/CultureInfoConverter.cs#L68-L130](https://github.com/dotnet/runtime/blob/5711ca33f8dd1758e18c09ebdedf7e95eb424e0f/src/libraries/System.ComponentModel.TypeConverter/src/System/ComponentModel/CultureInfoConverter.cs#L68-L130)

Reply:
It's quite a lot of code to inline & it seems like it predates the parsing APIs on `CultureInfo`. Tarek recommends using `GetCultureInfo` https://github.com/dotnet/runtime/pull/84154#discussion_r1158780580 which seems reasonable. I'd be curious if user feedback indicates that there are gaps.

Reply:
> It's quite a lot of code to inline 

Why not use CultureInfoConverter directly in this case?

Reply:
Looking more, it looks like there is a logic there to create a culture from display names too. GetCultureInfo is not enough to handle that. I agree with @eerhardt would be good just to use the CultureInfoConverter or to mimic the behavior from there.